packages/gatsby-plugin-google-tagmanager/src/gatsby-ssr.js

```diff
@@ -0,0 +1,37 @@
+import React from "react"
+import { stripIndent } from "common-tags"
+
+exports.onRenderBody = (
+  { setHeadComponents, setPreBodyComponents },
+  pluginOptions
+) => {
+  if (process.env.NODE_ENV === `production`) {
+    setHeadComponents([
+      <script
+        dangerouslySetInnerHTML={{
+          __html: stripIndent`
+            <!-- Google Tag Manager -->
+            <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+            new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+            j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+            'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+            })(window,document,'script','dataLayer', ${pluginOptions.id});</script>
+            <!-- End Google Tag Manager -->`,
+        }}
+      />,
+    ])
+
+    setPreBodyComponents([
+      <script
+        dangerouslySetInnerHTML={{
```

Comment:
Well... Gatsby *is* a **static site** generator. You could think of the React part as a (very) elaborate optimization.